src/idpi/operators/regrid.py

```diff
@@ -0,0 +1,184 @@
+            ymin=geo["latitudeOfFirstGridPointInDegrees"],
+            ymax=geo["latitudeOfLastGridPointInDegrees"],
+        )
+        if abs(obj.dx - geo["iDirectionIncrementInDegrees"]) > 1e-5:
+            raise ValueError("Inconsistent grid parameters")
+        if abs(obj.dy - geo["jDirectionIncrementInDegrees"]) > 1e-5:
+            raise ValueError("Inconsistent grid parameters")
+        return obj
+
+    @classmethod
+    def parse_regrid_operator(cls, op: str):
+        """Parse fieldextra out_regrid_target string.
+
+        Parameters
+        ----------
+        op : str
+            fieldextra out_regrid_target definition
+            i.e. crs,xmin,ymin,xmay,ymax,dx,dy.
+
+        """
+        crs_str, *grid_params = op.split(",")
+        crs = CRS.from_string(CRS_ALIASES[crs_str])
+        xmin, ymin, xmax, ymax, dx, dy = map(float, grid_params)
+        nx = (xmax - xmin) / dx + 1
+        ny = (ymax - ymin) / dy + 1
+        if nx != int(nx) or ny != int(ny):
+            raise ValueError("Inconsistent regrid parameters")
+        return cls(crs, int(nx), int(ny), xmin, xmax, ymin, ymax)
+
+    @property
+    def dx(self) -> float:
+        return (self.xmax - self.xmin) / (self.nx - 1)
+
+    @property
+    def dy(self) -> float:
+        return (self.ymax - self.ymin) / (self.ny - 1)
+
+    @property
+    def transform(self) -> transform.Affine:
+        return transform.from_origin(
+            west=self.xmin - self.dx / 2,
+            north=self.ymax + self.dy / 2,
+            xsize=self.dx,
+            ysize=self.dy,
+        )
+
+
+def regrid(field: xr.DataArray, dst: RegularGrid, resampling: Resampling):
+    """Regrid a field.
+
+    Parameters
+    ----------
+    field : xarray.DataArray
+        Input field defined on a regular grid in rotated latlon coordinates.
+    dst : RegularGrid
+        Destination grid onto which to project the field.
+    resampling : Resampling
+        Resampling method, alias of rasterio.warp.Resampling.
+
+    Raises
+    ------
+    ValueError
+        If the input field is not defined on a regular grid in rotated latlon or
+        if the input field geography metadata does not have consistent grid parameters.
+
+    Returns
+    -------
+    xarray.DataArray
+        Field regridded in the destination grid.
+
+    """
+    src = RegularGrid.from_field(field)
+
+    def reproject_layer(field):
+        output = np.zeros((dst.ny, dst.nx))
+        warp.reproject(
+            source=field[::-1],
+            destination=output,
+            src_crs=src.crs,
+            src_transform=src.transform,
+            dst_crs=dst.crs,
+            dst_transform=dst.transform,
+            resampling=resampling,
+        )
+        return output[::-1]
+
+    return xr.apply_ufunc(
+        reproject_layer,
+        field,
+        input_core_dims=[["y", "x"]],
+        output_core_dims=[["y1", "x1"]],
+        vectorize=True,
+    ).rename({"x1": "x", "y1": "y"})
```

Comment:
Why don't you `output_core_dims=[["y", "x"]],` and skip the renaming?
In case this is a workaround for xr.apply_ufunc's inability to have overlapping input and output core dims, maybe document that this is a workaround.